analyzere/extras/visualizations.py

```diff
@@ -0,0 +1,272 @@
+                self._format_MoneyField(layer.aggregate_limit))
+
+        # AggregateQuotaShare
+        if hasattr(layer, 'aggregate_period'):
+            if layer.aggregate_period > 0.0:
+                terms += '\nagg_period={}'.format(layer.aggregate_period)
+        if hasattr(layer, 'aggregate_reset'):
+            if layer.aggregate_reset > 1:
+                terms += '\nagg_reset={}'.format(layer.aggregate_reset)
+
+        # SurplusShare
+        if hasattr(layer, 'sums_insured'):
+            terms += '\nsums_insured={}'.format(
+                self._format_MoneyField(layer.sums_insured))
+        if hasattr(layer, 'retained_line'):
+            terms += '\nretained_line={}'.format(
+                self._format_MoneyField(layer.retained_line))
+        if hasattr(layer, 'number_of_lines'):
+            if layer.number_of_lines:
+                terms += '\nnumber_of_lines={}'.format(layer.number_of_lines)
+
+        # IndustryLossWarranty
+        if hasattr(layer, 'trigger'):
+            terms += '\ntrigger={}'.format(
+                self._format_MoneyField(layer.trigger))
+        if hasattr(layer, 'payout'):
+            terms += '\npayout={}'.format(
+                self._format_MoneyField(layer.payout))
+
+        # CatXL, IndustryLossWarranty
+        if hasattr(layer, 'nth'):
+            terms += '\nnth={}'.format(layer.nth)
+
+        # NoClaimsBonus
+        if hasattr(layer, 'payout_date'):
+            terms += '\npayout_date={}'.format(
+                self._format_DateField(layer.payout_date))
+        if hasattr(layer, 'payout_amount'):
+            terms += '\npayout={}'.format(
+                self._format_MoneyField(layer.payout_amount))
+
+        return terms
+
+    def _get_tree(self, l, include_terms):
+
+        if l.type == 'NestedLayer':
+            # hash the current node to see if it is unique
+            node_hash = hashlib.md5(str(l).encode('utf-8')).hexdigest()
+            if node_hash not in self._unique_nodes:
+                self._unique_nodes[node_hash] = self._sequence()
+
+            terms = self._format_layer_terms(l.sink) if include_terms else ''
+            name = ('Nested {} {}'.format(
+                    l.sink.type, self._format_description(l.description))
+                    if l.description else
+                    'Nested {} {}'.format(
+                    l.sink.type, self._format_description(l.sink.description))
+                    if l.sink.description else
+                    'Nested {} ({}) {}'
+                    .format(l.sink.type, self._unique_nodes[node_hash],
+                            terms))
+            for source in [self._get_tree(s, include_terms)
+                           for s in l.sources]:
+                if not (source, name) in self._edges:
+                    self._graph.attr('node', shape='ellipse',
+                                     style='transparent', color='black')
+                    self._graph.node(name)
```

Comment:
I will investigate the use of  `name` and  `label` and possibly `attr` and see.  We generate the name value such that it is either unique (even when there are duplicate nodes with the same name, by adding sequence, -OR- the hashing that we do ensures that we only have unique values).